components/search/HotelResults/HotelSelect.tsx

```diff
@@ -41,97 +41,121 @@ const HotelSelect: React.FC<HotelSelectUICompleteProps> = ({
+            availability: { available: true, minRoomsLeft: 10 },
+            reviews: { score: 4, totalReviews: 100, starRating: 4 },
+            price: { amount: 100, currency: { code: "USD", symbol: "$" } },
+          },
+          {
+            region_id: "3",
+            hotel_id: "3",
+            name: "Hotel 3",
+            image: { description: "img", url: "", alt: "alt" },
+            coordinates: { lat: 0, long: 0 },
+            availability: { available: true, minRoomsLeft: 10 },
+            reviews: { score: 4, totalReviews: 100, starRating: 4 },
+            price: { amount: 100, currency: { code: "USD", symbol: "$" } },
+          },
+        ],
+        summary: {
+          matchedPropertiesSize: 3,
+        },
+      });
+      setLoading(false);
+    }, 5000);
+  };
+
+  // Only re-renders on initial load and region change.
+  // We don't want to call the API after clicking one checkbox.
+  // So, in the future we could have an "apply new filters button" if filters change.
+  // Also, store filters in searchParams or localStorage...
+  useEffect(() => {
+    if (!isValid) return;
 
     if (process.env.NODE_ENV === "production") {
       handleFindHotels();
     } else {
-      setLoading(true);
-      setTimeout(() => {
-        setHotelsData({
-          priceRange: {
-            maxPrice: 5000,
-            minPrice: 0,
-          },
-          properties: [
-            {
-              region_id: "1",
-              hotel_id: "1",
-              name: "Hotel 1",
-              image: { description: "img", url: "", alt: "alt" },
-              coordinates: { lat: 0, long: 0 },
-              availability: { available: true, minRoomsLeft: 10 },
-              reviews: { score: 4, totalReviews: 100, starRating: 4 },
-              price: { amount: 100, currency: { code: "USD", symbol: "$" } },
-            },
-            {
-              region_id: "2",
-              hotel_id: "2",
-              name: "Hotel 2",
-              image: { description: "img", url: "", alt: "alt" },
-              coordinates: { lat: 0, long: 0 },
-              availability: { available: true, minRoomsLeft: 10 },
-              reviews: { score: 4, totalReviews: 100, starRating: 4 },
-              price: { amount: 100, currency: { code: "USD", symbol: "$" } },
-            },
-            {
-              region_id: "3",
-              hotel_id: "3",
-              name: "Hotel 3",
-              image: { description: "img", url: "", alt: "alt" },
-              coordinates: { lat: 0, long: 0 },
-              availability: { available: true, minRoomsLeft: 10 },
-              reviews: { score: 4, totalReviews: 100, starRating: 4 },
-              price: { amount: 100, currency: { code: "USD", symbol: "$" } },
-            },
-          ],
-          summary: {
-            matchedPropertiesSize: 3,
-          },
-        });
-        setLoading(false);
-      }, 5000);
+      handleFindHotelsCached();
     }
   }, [region]);
```

Comment:
With this approach, useEffect needs to have a dependency on both functions which are set on each render. (this is cautioned against).